crates/compilers/src/compile/project.rs

```diff
@@ -202,17 +227,17 @@ impl<'a, T: ArtifactOutput<CompilerContract = C::CompilerContract>, C: Compiler>
     ///   - check cache
     fn preprocess(self) -> Result<PreprocessedState<'a, T, C>> {
         trace!("preprocessing");
-        let Self { edges, project, mut sources, primary_profiles } = self;
+        let Self { edges, project, mut sources, primary_profiles, preprocessor } = self;
 
         // convert paths on windows to ensure consistency with the `CompilerOutput` `solc` emits,
         // which is unix style `/`
         sources.slash_paths();
 
-        let mut cache = ArtifactsCache::new(project, edges)?;
+        let mut cache = ArtifactsCache::new(project, edges, preprocessor.is_some())?;
```

Comment:
do the cache changes apply to any preprocessor that modifies sources or specifically to the one implemented in this PR? If the latter, this should be `.as_ref().is_some_and(|p| p.some_method())`

Reply:
it should be to any preprocessor that modifies sources, granted rn there's only one but others could be added